frontend/src/pages/Landing/index.js

```diff
@@ -250,6 +299,36 @@ function Landing() {
     fetchReports();
   }, [sortConfig, offset, perPage, alertsSortConfig, alertsOffset, alertsPerPage]);
 
+  // When reports are updated, make sure all checkboxes are unchecked
+  useEffect(() => {
+    setReportCheckboxes(makeReportCheckboxes(reports, false));
+  }, [reports]);
+
+  useEffect(() => {
+    const checkValues = Object.values(reportCheckboxes);
+    if (checkValues.every((v) => v === true)) {
+      setAllReportsChecked(true);
+    } else if (allReportsChecked === true) {
+      setAllReportsChecked(false);
+    }
+  }, [reportCheckboxes]);
+
+  const handleDownloadClick = () => {
+    const toDownloadableReportIds = (accumulator, entry) => {
+      if (!reports) return accumulator;
+      const [key, value] = entry;
+      if (value === false) return accumulator;
```

Comment:
I was going to try to exclude legacy reports early in this function, but was getting errors when trying to lookup which reports had legacyId values. I believe `reports.find` was throwing errors, so either it was a useState issue, or a scope/closure issue.

I stopped digging into it because the backend endpoint already filters out reports with `legacy` data, so excluding those ids here was just going to be a bonus anyway.